extensions/persistence.md

```diff
@@ -0,0 +1,160 @@
+
+The `PERSISTENCE` command is the primary means of communicating persistence settings. It has three subcommands. Implementations SHOULD ignore unrecognized subcommands, to be compatible with future extensions of this specification.
+
+### `PERSISTENCE STATUS`
+
+The `STATUS` subcommand is sent from the server to the client:
+
+    PERSISTENCE STATUS <client-setting> <effective-setting>
+
+`<client-setting>` has three possible values:
+
+* `ON`, indicating that the client has requested that persistence be enabled
+* `OFF`, indicating that the client has requested that persistence be disabled
+* `DEFAULT`, indicating that the client has no preference
+
+`<effective-setting>` has two possible values:
+
+* `ON`, indicating that persistence is in fact enabled
+* `OFF`, indicating that persistence is in fact disabled
+
+If the client is authenticated and the `draft/persistence` capability is negotiated, servers MUST send a `PERSISTENCE STATUS` command to the client as part of the registration burst, after the last `005` numeric but before either `376 RPL_ENDOFMOTD` or `422 ERR_NOMOTD`. This allows the client to adjust their behavior accordingly. If the `draft/persistence` capability is negotiated and an event occurs asynchronously that changes the persistence status (e.g. a server configuration change or `PERSISTENCE SET` from another client connection), servers SHOULD send `PERSISTENCE STATUS` to inform the client.
+
+If persistence is enabled, the server MUST send a `JOIN` line for each channel the client is joined to, subsequent to sending either `376` or `422`.
+
+### `PERSISTENCE GET`
+
+The `GET` subcommand is sent from the client to the server:
+
+    PERSISTENCE GET
+
+The server replies with a `PERSISTENCE STATUS` command as specified above.
+
+### `PERSISTENCE SET`
+
+The `SET` subcommand is sent from the client to the server:
+
+    PERSISTENCE SET <client-setting>
+
+where `<client-setting>` takes one of the three values specified above; the server replies with a `PERSISTENCE STATUS` command as specified above.
+
+#### Errors and Warnings
+
+Servers MUST NOT allow unauthenticated clients to become persistent, since there is no way for the client to reconnect in that case. Unauthenticated clients attempting to issue `GET` or `SET` MUST receive:
+
+    FAIL PERSISTENCE ACCOUNT_REQUIRED :An account is required
+
+On invalid parameters, the server MUST return:
+
+    FAIL PERSISTENCE INVALID_PARAMETERS :Invalid parameters
+
+If persistence cannot be accessed or modified due to an error, the `INTERNAL_ERROR` error code SHOULD be returned.
+
+    FAIL PERSISTENCE INTERNAL_ERROR :An error occurred
+
+Finally, servers MAY accept the change in client setting while not changing the effective setting. For example, persistence may be mandatory for all users, or it may be reserved for privileged users. In this event, the server MUST NOT send a `FAIL` response, but MUST send a `PERSISTENCE STATUS` response describing the current settings.
```

Comment:
What if servers want to send a human-readable message to explain why they don't apply `PERSISTENCE SET`?

Reply:
`WARN` or `NOTE` seem suitable for this purpose.